dsgrid/registry/project_registry_manager.py

```diff
@@ -302,3 +358,22 @@ def remove(self, config_id):
         self._remove(config_id)
         for key in [x for x in self._projects if x.id == config_id]:
             self._projects.pop(key)
+
+    def _remap_dimension_columns(self, df, mapping_references):
+        # This will likely become a common activity when running queries.
+        # May need to cache the result. But that will likely be in Project, not here.
+        # This method could be moved elsewhere.
+        for ref in mapping_references:
+            column = ref.from_dimension_type.value
+            if column in df.columns:
+                mapping_config = self._dimension_mapping_mgr.get_by_id(
+                    ref.mapping_id, version=ref.version
+                )
+                records = models_to_dataframe(mapping_config.model.records)
+                df = (
+                    df.join(records, getattr(df, column) == records.from_id)
+                    .drop("from_id")
+                    .drop(column)
+                    .withColumnRenamed("to_id", column)
```

Comment:
And here we're just switching to using to_id as our identifier. It could be that the mapping represents an aggregation; I agree that for this use case we don't care/don't want to actually deal with duplicates, but if/when this moves elsewhere we'll probably want to encode somewhere (the mapping_config?) whether we want to perform a sum or some other aggregation to collapse duplicates when we are done.

Reply:
Yes agreed Elaine.

Reply:
@elainethale I don't think this comment is relevant to this PR (or at least there are no changes associated with it), so I'm resolving it. I'll assume that you or Meghan will make appropriate plans for any changes to mapping configs.